src/game/characters/BaseCharacter.ts

```diff
@@ -0,0 +1,34 @@
+export interface BaseStats {
+    health: number;
+    attack: number;
+    defense: number;
+    speed: number;
+}
+
+export class BaseCharacter {
+    protected stats: BaseStats;
```

Comment:
don't think we should protect this, as stats will be mutated (I'm thinking of hp for example)